src/poetry/factory.py

```diff
@@ -126,10 +134,15 @@ def create_config(cls, io: Optional["IO"] = None) -> Config:
         return config
 
     @classmethod
-    def configure_sources(
-        cls, poetry: Poetry, sources: List[Dict[str, str]], config: Config, io: "IO"
-    ) -> None:
-        for source in sources:
+    def create_pool(cls, config: "Config", io: Optional["IO"] = None) -> "Pool":
+        from poetry.repositories.pool import Pool
```

Comment:
This delayed import really seems useless -- the import will run almost right away anything. I think this pattern tends to be a premature optimization that is hard to maintain and that we should address import overhead with a higher-level/more unified approach if it's an issue.